### DISCUSSION TITLE
Hourly SNOTEL data through fetchSCAN

### DISCUSSION THREAD

**Discussion by morrismc99** [ncss-tech/soilDB#260] [Q&A] [answered]
Hello, I wanted to inquire about the functionality of using fetchSCAN to retrieve hourly SNOTEL site data. I've been very successful using fetchSCAN to deliver daily data from a SNOTEL site of interest; however, I cannot seem to access the hourly data. I know this site collects hourly T, precip, SWE data, but I had hoped to be able to use fetchSCAN to access it in a convenient way.

Older documentation of this function indicates you can input "timeseries = 'hourly'" however that's been unsuccessful in attempts at yielding useful data. 

Any insights welcome. 
Thanks so much,
Matthew

**Comment by dylanbeaudette** [comment]
Hi Matthew,

Thanks for asking. I'll check into it today and report back. It should be possible to return hourly data.

**Comment by brownag** [comment, marked as answer]
UPDATE:
now in development version you should be able to supply just the timeseries argument and get batched results.

For instance:
```r
x <- fetchSCAN(2072, 2010:2020, timeseries='Hourly')
```

---

There is still support for the old form of the function allowing arbitrary list of request parameters to be posted to form via `req` argument; but this is not documented. In current soilDB for the backward compatible interface that `req` argument needs to be _named_, and use of it will trigger a deprecation message. 

`req` argument was deprecated primarily due to the fact that the newer interface changed first required arguments which are now a vector of `site.code`/`year`. These newer arguments facilitate batching/combining results from multiple "reports", but that functionality was not extended for all possible options that the original, completely generic, `req` interface allowed for (i.e. hourly data)

So, in the current version you can do something like this (with a deprecation warning):

```r
x <- fetchSCAN(req = c(intervalType=' View Historic ', report='SMS', timeseries='Hourly', 
                       format='copy', sitenum='2072', interval='YEAR', year='2011', month='CY'))
```


I made a small change that allows form variables to be specified as additional arguments (via `...`). 

Now in development version (after bf8b6daa) you can equivalently do this:

```r
x <- fetchSCAN(intervalType=' View Historic ', report='SMS', timeseries='Hourly',
               format='copy', sitenum='2072', interval='YEAR', year='2011', month='CY')
```

I think this is what you were hoping would work i.e. timeseries='Hourly' can be specified directly as an argument.

Note that for now you will need to manually iterate over sites/years with either syntax and then rbind the data.frame results.

It will take a little bit more effort to get the generic request parameters working in the batched interface, but that shouldn't be too difficult to implement. I have added this as a TODO in the "fetchSCAN() upgrades" issue https://github.com/ncss-tech/soilDB/issues/184

**Reply by morrismc99** [reply]
This is awesome! And exactly the type of functionality I was looking for. Thanks again for you help on this ! 🙏

**Reply by brownag** [reply]
Great, now in development version you should be able to supply just the `timeseries` argument and get batched results. 

For instance: 
```r
x <- fetchSCAN(2072, 2010:2020, timeseries='Hourly')
```
